tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -276,6 +259,31 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		// try to create an IDGraph if there's none - `who` will be the prime identity
+		// please note the web3networks for the prime identity will be all avaiable networks
+		fn maybe_create_id_graph(who: &Identity) -> Result<(), DispatchError> {
```

Comment:
Will this newly created identity show up in DI response ?

Reply:
It will.

Imagine you link an identity A to your prime identity, then both your prime identity and A will show up in the `mutated_id_graph` because that's the exactly the IDGraph change (it was empty before)

Reply:
I see so this means id graph is created but prime identity is deactivated - thanks.

Reply:
But in case of `set_identity_networks` trusted call there is only one identity pushed to `mutated_id_graph` while this logic may alter two identities sometimes.

```
				let mut mutated_id_graph = IDGraph::<Runtime>::default();
				if let Some(identity_context) = IMT::id_graphs(&who, &identity) {
					if let Some(key) = maybe_key {
						mutated_id_graph.push((identity, identity_context));
						return Ok(TrustedCallResult::SetIdentityNetworks(
							SetIdentityNetworksResult {
								mutated_id_graph: aes_encrypt_default(
									&key,
									&mutated_id_graph.encode(),
								),
								id_graph_hash,
							},
						))
					}
				} else {
					// if should not happen, so we just log the error here
					error!("failed to get identity_context for {:?}, {:?}", &who, &identity);
				}
```

Reply:
> this set_identity_networks call may alter two identities sometimes

How could this happen? 😂

Reply:
Create prime identity [here]( https://github.com/litentry/litentry-parachain/pull/2347/files/c45491a137d12cf9894127d18b4d81eaff74f606#diff-0673e79aa46adac84174d588afcb0816746db60d4b560b4cdd01097951b322b0R209)
Mutate identity [here](https://github.com/litentry/litentry-parachain/pull/2347/files/c45491a137d12cf9894127d18b4d81eaff74f606#diff-0673e79aa46adac84174d588afcb0816746db60d4b560b4cdd01097951b322b0R212)

? 😕

Reply:
If you are setting the networks of the prime identity for the first time - the IDGraph must have not existed;
if you are setting the networks of a non-prime identity, then either the IDGraph must have existed or it fails because identity doesn't exist (for an empty IDGraph)

So in either case you can't alter two identities?

Reply:
But I think what's missing is for the second case, should the mutated_id_graph contain the prime identity? The extrinsic in tee-pallet is unfortunately not atomic

Reply:
Adding `edit_block` to `IdentityContext` would make it possible to detect all changes inside idgraph at `trusted_call` level and have common and accurate method for producing idgraph diff.

Reply:
> If you are setting the networks of the prime identity for the first time - the IDGraph must have not existed; if you are setting the networks of a non-prime identity, then either the IDGraph must have existed or it fails because identity doesn't exist (for an empty IDGraph)
> 
> So in either case you can't alter two identities?

Ok I see - it's not clear by looking just at this code.

Reply:
> Adding edit_block to IdentityContext

Do you mean look for any entry whose `edit_block` equals to the current block number?
Changing IdentityContext struct might need a storage migration (or maybe not if you append it)

Can we do a simple diff? - it's keyed by the prime identity so the computation shouldn't be that costly

Reply:
> Do you mean look for any entry whose edit_block equals to the current block number?

Exactly 🙃 

> Can we do a simple diff? - it's keyed by the prime identity so the computation shouldn't be that costly

Should do the job also without extra storage space.

Reply:
Implemented - re-request the review now @kziemianek :)

Reply:
Thanks 👍🏼